WooCommerce/Classes/POS/Presentation/CardReaderConnection/UI States/CardPresentPaymentBluetoothRequiredAlertView.swift

```diff
@@ -0,0 +1,30 @@
+import SwiftUI
+
+struct CardPresentPaymentBluetoothRequiredAlertView: View {
+    private let viewModel: CardPresentPaymentBluetoothRequiredAlertViewModel
+    @Environment(\.dismiss) var dismiss
+
+    init(viewModel: CardPresentPaymentBluetoothRequiredAlertViewModel) {
+        self.viewModel = viewModel
+    }
+
+    var body: some View {
+        VStack {
+            Text(viewModel.title)
+
+            viewModel.image
+
+            Text(viewModel.errorDetails)
+
+            Button(viewModel.openSettingsButtonViewModel.title,
+                   action: viewModel.openSettingsButtonViewModel.actionHandler)
+            .buttonStyle(PrimaryButtonStyle())
+
+            Button(viewModel.dismissButtonViewModel.title) {
+                dismiss()
+                viewModel.dismissButtonViewModel.actionHandler()
+            }
+            .buttonStyle(SecondaryButtonStyle())
+        }
+    }
+}
```

Comment:
Might be worth having a preview for this? I tried making a button with sheet content to test out the dismiss action in preview, but somehow it didn't work.

```swift
#Preview {
    @State var showsSheet = false
    return Button("Show sheet", action: {
        showsSheet = true
    })
        .sheet(isPresented: $showsSheet) {
            CardPresentPaymentBluetoothRequiredAlertView(viewModel: .init(error: NSError(domain: "", code: 1), endSearch: {}))
        }
}
```


```suggestion
}

#Preview {
    CardPresentPaymentBluetoothRequiredAlertView(viewModel: .init(error: NSError(domain: "", code: 1), endSearch: {}))
```

Reply:
I guess that the dismiss environment variable perhaps doesn't get set in Previews? I've never tried that. But yes, thanks for adding the preview...